packages/firestore/src/api/database.ts

```diff
@@ -458,6 +480,8 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
       this._config.credentials,
       this._queue
     );
+
+    this.isClientRunning = true;
```

Comment:
I'd move this line up, so that if we ever add any async code above, the flag will still be set to true even if the user manages to sneak in a call to `clearPersistence()` while we are waiting for an async operation to complete.